app/src/main/java/com/navapp/navigation/route/RouteFactory.java

```diff
@@ -0,0 +1,11 @@
+package com.navapp.navigation.route;
+
+import com.google.maps.PendingResult;
+import com.navapp.navigation.destination.Destination;
+import com.navapp.navigation.destination.data.Location;
+
+import java.util.List;
+
+public interface RouteFactory {
+    void callbackCreate(Location start, List<Destination> destinations, Location end, PendingResult.Callback<Route> callback);
```

Comment:
Terrible name.... Makes it sound like the purpose of the method is to create a callback.
If you're looking for some work to specify that this is an async api, then using something like:
- `createAsync`
- `requestCreation`
- `startCreation`
- etc

Reply:
yeah, it didn't feel right